### DISCUSSION TITLE
Change to EmailSubmissions functions

### DISCUSSION THREAD

**Discussion by M-HFM** [linagora/tmail-flutter#2230] [Q&A]
Hi,
I'm trying to make some changes in the project to be compatible with sending emails with my jmap server, I'm having some difficulty finding the dart files responsible for sending e-mails and the outbox creation, could you help me indicating with and where are locate those files and other that could be related to this function?

**Comment by hoangdat** [comment]
Hi, can you share what the problem you have? And what kind of server you are using.

- We have a use-case `SendEmailInteractor` to handle the logic of sending email
- And `Email` is managed by `EmailRepository` which have `sendEmail` method.
<img width="412" alt="image" src="https://github.com/linagora/tmail-flutter/assets/6462404/6f8146d2-ccb2-4c96-8198-88fb25783e9b">

- and `EmailAPI` is JMAP implementation
- btw, we also did a PR for supporting Fastmail, you can check it: https://github.com/linagora/tmail-flutter/pull/974

Thanks and BRs.

**Reply by M-HFM** [reply]
Hi thanks a lot for the feedback,
I'm using stalwartlabs mail server (https://github.com/stalwartlabs/mail-server), and I get an error when trying to send emails (as shown on the image), the 
message go to the outbox folder but the sending process is not complete. The outbox folder created and the messages on it are shown in other clients like mailtemi and Thunderbird.
![testsubmission](https://github.com/linagora/tmail-flutter/assets/32017845/a478dc10-626c-4a95-abc8-31849066f407)

**Reply by M-HFM** [reply]
Besides this problem with sending emails, the other functions of the app are working properly with the server

**Reply by hoangdat** [reply]
oh, so great to hear that. 

We also have a pending PR for a long time https://github.com/linagora/tmail-flutter/pull/1850, which also related to `EmailSubmission` also. 

And we are now on a tight time to deliver the roadmap, can you help us to solve this problem with your server. 

Thanks and BRs.

**Reply by hoangdat** [reply]
hi @M-HFM , can you show me the JMAP response  of sending email?

**Reply by M-HFM** [reply]
Hi,Thanks for the feedback how can I provide you the necessary data?

**Comment by hoangdat** [comment]
Hi @M-HFM , nice to say to you we are in progress of supporting stalwart: https://github.com/linagora/tmail-flutter/pull/2315

**Comment by M-HFM** [comment]
Hi, Thanks a lot, I'll follow this and continuing testing and reporting how its working with my server Atenciosamente / Best regards / Cordialement, Mateus H. F. Mello. On Nov 7, 2023, at 11:08 AM, Dat H. Pham ***@***.***> wrote: Hi @M-HFM , nice to say to you we are in progress of supporting stalwart: #2315 — Reply to this email directly, view it on GitHub , or unsubscribe . You are receiving this because you were mentioned. Message ID: ***@***.***>

**Comment by chibenwa** [comment]
Support for Stallwart is merged: do not hesitate to have a look and report issues ;-)
